src/components/Header/Header.tsx

```diff
@@ -16,22 +20,33 @@ const BackgroundStyle = {
 
 export const Header = () => {
     const [isOpenModal, setIsOpenModal] = React.useState(false);
+    const {activeMovie, setActiveMovie} = React.useContext(MovieContext);
+
+    const getMovieById = React.useMemo(() => {
+       return MOVIES_LIST.find(item => item.id === activeMovie);
+    },[activeMovie]);
 
     return (
         <header className={styles.header} style={BackgroundStyle}>
             <div className={styles.content}>
-                <Logo />
-                <Button className={styles.button} name='+ Add Movie' onClick={() => setIsOpenModal(true)} />
-                <Modal open={isOpenModal} onClose={() => setIsOpenModal(false)}>
-                    <>
-                        <h1 className={styles.title}>Add Movie</h1>
-                        <React.Suspense fallback='loading...'>
-                            <AddMovieForm />
-                        </React.Suspense>
-                    </>
-                </Modal>
+                <Logo onClick={() => setActiveMovie(null)} />
+                {
+                    activeMovie ?
+                        <SearchIcon className={styles.search} onClick={() => setActiveMovie(null)} /> :
+                        <>
+                            <Button className={styles.button} name='+ Add Movie' onClick={() => setIsOpenModal(true)} />
+                            <Modal open={isOpenModal} onClose={() => setIsOpenModal(false)}>
+                                <>
+                                    <h1 className={styles.title}>Add Movie</h1>
+                                    <React.Suspense fallback='loading...'>
+                                        <AddMovieForm />
+                                    </React.Suspense>
+                                </>
+                            </Modal>
+                        </>
+                }
             </div>
-            <SearchBlock />
+            {activeMovie && getMovieById ? <MovieDetails {...getMovieById} /> : <SearchBlock />}
```

Comment:
I think that you don’t need `getMovieById` here. Cause it's function and will be always `true`

Reply:
I won't be able to remove the function because it might return undefined and I won't be able to use the spread operator